### PR TITLE
Add route files creation

### PR DESCRIPTION
**Pull request by tamorim**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,17 +3,42 @@ import gutil from 'gulp-util'
 import through from 'through2'
 import extractImports from './extract-imports'
 
-function createMetadataFile (file, imports) {
-  const base = path.join(file.base, '..', 'components')
-  const filePath = gutil.replaceExtension(path.join(base, file.relative), '.json')
-
+function createFile (base, filePath, content) {
```
Comment (brenoc) [inline src/index.js]: Maybe rename to `createJsonFile`? Since we are doing a JSON.stringify in it's content.

File: src/index.js [hunk 2]
```diff
@@ -3,17 +3,42 @@ import gutil from 'gulp-util'
...
 }
 
+function createMetadataFile (file, imports) {
+  const base = path.join(file.base, '..', 'components')
+  const filePath = gutil.replaceExtension(path.join(base, file.relative), '.json')
+
+  return createFile(base, filePath, imports)
+}
+
+function createRouteFile (file, manifest, route) {
```
Comment (brenoc) [inline src/index.js]: Why do you need `manifest`?

File: src/index.js [hunk 3]
```diff
@@ -3,17 +3,42 @@ import gutil from 'gulp-util'
...
+    path: route.path,
+  }
+
+  return createFile(base, filePath, content)
+}
+
+function createRouteSettings (file, manifest, route) {
+  const base = path.join(file.base, '..', 'settings', 'routes')
+  const filePath = path.join(base, `${route.name}.json`)
+  const componentPath = file.path.replace(/^.*render\/assets\//, '')
```
Comment (brenoc) [inline src/index.js]: I think it's safer to use https://nodejs.org/api/path.html#path_path_sep here

**Comment by brenoc** [comment]
Tests are missing

**Comment by tamorim** [comment]
@brenoc Apart from the tests, do you think that everything is okay?

**Comment by firstdoit** [comment]
Nice! LGTM

**Pull request merged**